src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -13,6 +13,7 @@
 using System.Management.Automation.Internal;
```

Comment:
@iSazonov, your last commit had 118 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

ConsoleHost unit tests.ShellInterop.Verify Validate Dollar Error Populated should throw exception
```powershell
Expected an exception, with FullyQualifiedErrorId 'CommandNotFoundException' to be thrown, but no exception was thrown.
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\ConsoleHost.Tests.ps1: line 109
109:             } | Should -Throw -ErrorId "CommandNotFoundException" -PassThru
```
ConsoleHost unit tests.CommandLine.array -args
```powershell
Cannot index into a null array.
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\ConsoleHost.Tests.ps1: line 134
134:             (& $powershell -noprofile { $args[1] } -args 1,(2,3))[1]  | Should -Be 3

```
ConsoleHost unit tests.CommandLine.-File should be default parameter
```powershell
Expected 'hello', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\ConsoleHost.Tests.ps1: line 165
165:             $observed | Should -Be "hello"
```
ConsoleHost unit tests.CommandLine.-File should pass additional arguments to script
```powershell
Expected 2, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\ConsoleHost.Tests.ps1: line 181
181:             $observed.Count | Should -Be 2
```
ConsoleHost unit tests.CommandLine.A single dash should be passed as an arg
```powershell
Expected exactly 'h-llo', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\ConsoleHost.Tests.ps1: line 248
248:             $observed | Should -BeExactly "h-llo"
```